Submitted by: Matt Burgess (matthew at linuxfromscratch dot org)
Date: 2009-05-03
Initial Package Version: 2.5.35
Origin: Matt Burgess
Upstream Status: Submitted (attached to sourceforge bug 2178663)
Description: Fixes an error caused by header cleanups in GCC 4.4.0 that is
             evident from the test suite and would affect any C++ lexers
             generated by Flex.  Without this patch, Flex will generate lexers
             containing references to the 'EOF' symbol without including the
             necessary C++ header file, leading to:

             error: 'EOF' was not declared in this scope 

diff -Naur flex-2.5.35.orig/skel.c flex-2.5.35/skel.c
--- flex-2.5.35.orig/skel.c	2008-02-26 21:34:19.000000000 +0000
+++ flex-2.5.35/skel.c	2009-05-03 15:18:14.000000000 +0000
@@ -284,6 +284,7 @@
   "/* begin standard C++ headers. */",
   "#include <iostream> ",
   "#include <errno.h>",
+  "#include <cstdio>",
   "#include <cstdlib>",
   "#include <cstring>",
   "/* end standard C++ headers. */",
